src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -62,108 +59,54 @@ interface SectionData {
   category: CategoryWithDapps
 }
 
-function mapDappFields(dapp: any, address: string, isFeatured: boolean): Dapp {
-  return {
-    id: dapp.id,
-    categoryId: dapp.categoryId,
-    name: dapp.name,
-    iconUrl: dapp.logoUrl,
-    description: dapp.description,
-    dappUrl: dapp.url.replace('{{address}}', address ?? ''),
-    isFeatured,
-  }
-}
-
 export function DAppsExplorerScreen() {
-  const { t, i18n } = useTranslation()
-  const dappsListUrl = useSelector(dappsListApiUrlSelector)
-  const address = useSelector(walletAddressSelector)
+  const { t } = useTranslation()
   const [isHelpDialogVisible, setHelpDialogVisible] = useState(false)
   const insets = useSafeAreaInsets()
   const scrollPosition = useRef(new Animated.Value(0)).current
   const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+  const dispatch = useDispatch()
+  const dappsList = useSelector(dappsListSelector)
+  const dappsCategories = useSelector(dappsCategoriesSelector)
+  const featuredDapp = useSelector(featuredDappSelector)
+  const loading = useSelector(dappsListLoadingSelector)
+  const error = useSelector(dappsListErrorSelector)
+  const [categoriesById, setCategoriesById] = useState<CategoryWithDapps[]>([])
 
   const { onSelectDapp, ConfirmOpenDappBottomSheet } = useOpenDapp()
 
-  const shortLanguage = i18n.language.split('-')[0]
-
   useEffect(() => {
+    dispatch(fetchDappsList())
     ValoraAnalytics.track(DappExplorerEvents.dapp_screen_open)
   }, [])
 
-  const {
-    loading,
-    error,
-    result,
-  }: {
-    loading: boolean
-    error: Error | undefined
-    result: { categories: CategoryWithDapps[]; featured: Dapp | undefined } | undefined
-  } = useAsync(
-    async () => {
-      if (!dappsListUrl) {
-        throw new Error('Dapps list url is not defined')
+  useEffect(() => {
+    const mappedCategories: { [id: string]: CategoryWithDapps } = {}
+    dappsCategories.forEach((cat: any) => {
+      mappedCategories[cat.id] = {
+        id: cat.id,
+        name: cat.name,
+        fontColor: cat.fontColor,
+        backgroundColor: cat.backgroundColor,
+        dapps: [],
       }
+    })
+    dappsList.forEach((dapp: Dapp) => {
+      mappedCategories[dapp.categoryId].dapps.push(dapp)
+    })
+    setCategoriesById(Object.values(mappedCategories))
+  }, [dappsList, dappsCategories])
```

Comment:
Looks like we should have a selector instead of this effect.

Reply:
yeah good call, i moved this to a selector and added a test!